lib/openstack_query/query_blocks/query_executer.py

```diff
@@ -29,7 +31,11 @@ def __init__(self, prop_enum_cls: Type[PropEnum], runner_cls: Type[RunnerWrapper
         self.runner = runner_cls(self._prop_enum_cls.get_marker_prop_func())
         self._client_side_filters = None
         self._server_side_filters = None
-        self._raw_results = []
+        self._results = []
+
+        # for chaining
+        self._forwarded_values = {}
```

Comment:
Hmm I'm not keen on this because now we're leaking the concept of chaining throughout our codebase

IMO, the chaining should be isolated and effectively run the executor in a way it can't tell if the user or chainer has invoked this path rather than it being a special case?

Reply:
so pass forwarded values as a parameter to `run_query`?
and store forwarded_values in `QueryChainer?`
The Query needs to store forwarded outputs somewhere so if the user calls `run()` again we need to evaluate it again